standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/DirectSqlUpdateStat.java

```diff
@@ -647,6 +653,12 @@ public long getNextCSIdForMPartitionColumnStatistics(long numStats) throws MetaE
       jdoConn = pm.getDataStoreConnection();
       dbConn = (Connection) (jdoConn.getNativeConnection());
 
+      if (sqlGenerator.getDbProduct().isMYSQL()) {
+        try (Statement stmt = dbConn.createStatement()) {
+          stmt.execute("SET @@session.sql_mode=ANSI_QUOTES");
+        }
+      }
+
```

Comment:
Can we move the two code blocks to a function and invoke the function in the two places.

Reply:
done